src/components/JsonViewer/JsonViewer.tsx

```diff
@@ -93,26 +89,25 @@ class Json extends React.PureComponent<JsonProps> {
     }
   };
 
-  clickListener = (event: MouseEvent) => {
-    this.collapseElement(event.target as HTMLElement);
-  };
+  const clickListener = React.useCallback((event: MouseEvent) => {
+    collapseElement(event.target as HTMLElement);
+  }, []);
 
-  focusListener = (event: KeyboardEvent) => {
+  const focusListener = React.useCallback((event: KeyboardEvent) => {
     if (event.key === 'Enter') {
-      this.collapseElement(event.target as HTMLElement);
+      collapseElement(event.target as HTMLElement);
     }
-  };
+  }, []);
 
-  componentDidMount() {
-    this.node!.addEventListener('click', this.clickListener);
-    this.node!.addEventListener('focus', this.focusListener);
-  }
+  React.useEffect(() => {
+    if (node) {
+      node?.addEventListener('click', clickListener);
+      node?.addEventListener('focus', focusListener);
+    }
+  }, [clickListener, focusListener, node]);
 
-  componentWillUnmount() {
-    this.node!.removeEventListener('click', this.clickListener);
-    this.node!.removeEventListener('focus', this.focusListener);
-  }
-}
```

Comment:
should't we remove event listener or I missed something?

Reply:
fixed. Thank for the comment.